example of a build error:

```
checking for /.../openwrt/staging_dir/host/bin/gcc option to enable large file support... -D_FILE_OFFSET_BITS=64
checking for /.../openwrt/staging_dir/host/bin/gcc option for timestamps after 2038... support not detected
configure: error: in '/.../openwrt/build_dir/host/findutils-4.10.0':
configure: error: could not enable timestamps after mid-January 2038.
This package recommends support for these later
timestamps. However, to proceed with signed 32-bit
time_t even though it will fail then, configure with
'--disable-year2038'.
See 'config.log' for more details
make[2]: *** [Makefile:38: /.../openwrt/build_dir/host/findutils-4.10.0/.configured] Error 1
```

--- a/include/host-build.mk
+++ b/include/host-build.mk
@@ -67,6 +67,10 @@ HOST_CONFIGURE_ARGS = \
 	--localstatedir=$(HOST_BUILD_PREFIX)/var \
 	--sbindir=$(HOST_BUILD_PREFIX)/bin
 
+ifneq ($(YEAR_2038),y)
+  HOST_CONFIGURE_ARGS += --disable-year2038
+endif
+
 HOST_MAKE_VARS = \
 	CFLAGS="$(HOST_CFLAGS)" \
 	CPPFLAGS="$(HOST_CPPFLAGS)" \

--- a/rules.mk
+++ b/rules.mk
@@ -26,6 +26,7 @@ qstrip=$(strip $(subst ",,$(1)))
 empty:=
 space:= $(empty) $(empty)
 comma:=,
+pound:=\#
 merge=$(subst $(space),,$(1))
 confvar=$(shell echo '$(foreach v,$(1),$(v)=$(subst ','\'',$($(v))))' | $(MKHASH) md5)
 strip_last=$(patsubst %.$(lastword $(subst .,$(space),$(1))),%,$(1))
@@ -376,6 +377,19 @@ endef
 
 define shexport
 export $(call shvar,$(1))=$$(call $(1))
+endef
+
+# Test support for 64-bit time with C code from largefile.m4 provided by GNU Gnulib
+# the value is 'y' when successful and '' otherwise
+define YEAR_2038
+$(shell \
+  mkdir -p $(TMP_DIR); \
+  echo '$(pound) include <time.h>' > $(TMP_DIR)/year2038.c; \
+  echo '$(pound) define LARGE_TIME_T ((time_t) (((time_t) 1 << 30) - 1 + 3 * ((time_t) 1 << 30)))' >> $(TMP_DIR)/year2038.c; \
+  echo 'int verify_time_t_range[(LARGE_TIME_T / 65537 == 65535 && LARGE_TIME_T % 65537 == 0) ? 1 : -1];' >> $(TMP_DIR)/year2038.c; \
+  echo 'int main (void) {return 0;}' >> $(TMP_DIR)/year2038.c; \
+  $(HOSTCC) $(TMP_DIR)/year2038.c -o /dev/null 2>/dev/null && echo y && rm -f $(TMP_DIR)/year2038.c || rm -f $(TMP_DIR)/year2038.c; \
+)
 endef
 
 # Execute commands under flock

--- a/tools/coreutils/Makefile
+++ b/tools/coreutils/Makefile
@@ -29,7 +29,6 @@ HOST_GNULIB_SKIP := \
 	lib/locale.in.h
 
 HOST_CONFIGURE_ARGS += \
-	--disable-year2038 \
 	 --enable-install-program=$(subst $(space),$(comma),$(strip $(PKG_PROGRAMS)))
 
 HOST_MAKE_FLAGS += \

